src/strategies/InterestStrategy.sol

```diff
@@ -0,0 +1,286 @@
+    error InvalidFeeTo();
+    error InvalidMaxInterestPerSecond();
+    error InvalidLerpParameters();
+
+    event LogAccrue(uint256 accruedAmount);
+    event LogInterestChanged(uint64 interestPerSecond);
+    event LogInterestWithLerpChanged(uint64 startInterestPerSecond, uint64 targetInterestPerSecond, uint64 duration);
+    event FeeToChanged(address previous, address current);
+    event SwapperChanged(address previous, address current);
+    event Swap(uint256 amountIn, uint256 amountOut);
+    event SwapTokenOutEnabled(IERC20 token, bool enabled);
+    event SwapAndWithdrawFee(uint256 amountIn, uint256 amountOut, IERC20 tokenOut);
+    event WithdrawFee(uint256 amount);
+    event EmergencyExitEnabled(bool enabled);
+
+    uint256 private constant WAD = 1e18;
+
+    // Interest linear interpolation to destination in a given time
+    // ex: 1% -> 13% in 30 days.
+    struct InterestLerp {
+        uint64 startTime;
+        uint64 startInterestPerSecond;
+        uint64 targetInterestPerSecond;
+        uint64 duration;
+    }
+    
+    // slot grouping
+    uint128 public pendingFeeEarned;
+    uint128 public pendingFeeEarnedAdjustement;
+
+    // slot grouping
+    uint64 public lastAccrued;
+    uint64 public interestPerSecond;
+    bool public emergencyExitEnabled;
+
+    address public feeTo;
+    address public swapper;
+    uint256 public principal;
+    mapping(IERC20 => bool) public swapTokenOutEnabled;
+    InterestLerp public interestLerp;
+
+    constructor(
+        IERC20 _strategyToken,
+        IERC20 _mim,
+        IBentoBoxV1 _bentoBox,
+        address _feeTo
+    ) BaseStrategy(_strategyToken, _bentoBox) {
+        feeTo = _feeTo;
+        swapTokenOutEnabled[_mim] = true;
+
+        emit FeeToChanged(address(0), _feeTo);
+        emit SwapTokenOutEnabled(_mim, true);
+    }
+
+    function getYearlyInterestBips() external view returns (uint256) {
+        return (interestPerSecond * 100) / 316880878;
+    }
+
+    function _updateInterestPerSecond() private {
+        if (interestLerp.duration == 0) {
+            return;
+        }
+
+        /// @dev Adapted from https://github.com/makerdao/dss-lerp/blob/master/src/Lerp.sol
+        if (block.timestamp < interestLerp.startTime + interestLerp.duration) {
+            uint256 t = ((block.timestamp - interestLerp.startTime) * WAD) / interestLerp.duration;
+            interestPerSecond = uint64(
+                (interestLerp.targetInterestPerSecond * t) /
+                    WAD +
+                    interestLerp.startInterestPerSecond -
+                    (interestLerp.startInterestPerSecond * t) /
+                    WAD
+            );
+        } else {
+            interestPerSecond = interestLerp.targetInterestPerSecond;
+            interestLerp.duration = 0;
+        }
+    }
+
+    function _skim(uint256) internal override {
+        principal = availableAmount();
+
+        if (principal > 0) {
+            lastAccrued = uint64(block.timestamp);
```

Comment:
where has the accrual happened here?

Reply:
i guess the skim should only be called after a harvest, still the lastAccrued seems wrong